js_modules/dagster-ui/packages/ui-core/src/launchpad/LaunchpadSession.tsx

```diff
@@ -628,77 +630,86 @@ const LaunchpadSession = (props: LaunchpadSessionProps) => {
           <>
             <LoadingOverlay isLoading={configLoading} message={LOADING_CONFIG_FOR_PARTITION} />
             <SessionSettingsBar>
-              <ConfigEditorConfigPicker
-                partitionSetDetails={partitionSetDetails}
-                pipeline={pipeline}
-                partitionSets={partitionSets.results}
-                base={currentSession.base}
-                onSaveSession={onSaveSession}
-                onSelectPreset={onSelectPreset}
-                onSelectPartition={onSelectPartition}
-                repoAddress={repoAddress}
-                assetSelection={currentSession.assetSelection}
-              />
-              <SessionSettingsSpacer />
-              {launchpadType === 'asset' ? (
-                <Box flex={{gap: 16, alignItems: 'center'}}>
-                  <TextInput
-                    readOnly
-                    value={
-                      currentSession.assetSelection
-                        ? currentSession.assetSelection
-                            .map((a) => tokenForAssetKey(a.assetKey))
-                            .join(', ')
-                        : '*'
-                    }
+              {!onSaveConfig ? (
```

Comment:
This ternary is very long and hard for humans to follow. (shakes fist at AI)

Reply:
yeah for real -- also seemingly maybe inaccurate (ended up inverting some of the conditions while refactoring).

moved stuff out into separate consts, which feels more readable to my eye, let me know if that feels better to you as well